src_c/camera_v4l2.c

```diff
@@ -29,42 +29,64 @@ char **
 v4l2_list_cameras(int *num_devices)
 {
     char **devices;
-    char *device;
-    int num, i, fd;
+    char *device = NULL;
+    int num, i, j, fd;
 
     num = *num_devices;
 
     devices = (char **)malloc(sizeof(char *) * 65);
+    if (!devices) {
+        return NULL;
+    }
 
     device = (char *)malloc(sizeof(char) * 13);
+    if (!device) {
+        goto error;
+    }
+
     strcpy(device, "/dev/video");
     fd = open(device, O_RDONLY);
     if (fd != -1) {
         devices[num] = device;
         num++;
         device = (char *)malloc(sizeof(char) * 13);
+        if (!device) {
+            goto error;
```

Comment:
should there be a `   close(fd);` before the `goto error;`?

Reply:
Good catch! I pushed another commit where I reworked the code, so it should be a bit simpler now, and this has been fixed